kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Gettable.java

```diff
@@ -23,4 +25,11 @@
    */
   T get();
 
+
+  /**
+   * @return the item or throws Exception if the item doesn't exist.
+   * @throws io.fabric8.kubernetes.client.KubernetesClientException if an error occurs
+   */
+  T getRequired() throws ResourceNotFoundException;
```

Comment:
getNotNull()? Guava has [checkNotNull()](http://guava-libraries.googlecode.com/svn-history/r14/trunk/javadoc/com/google/common/base/Preconditions.html#checkNotNull(T)) though thats more of a precondition check. how about getAndNotNull()?